Tests/PersonalityInsightsV3Tests/PersonalityInsightsTests.swift

```diff
@@ -0,0 +1,170 @@
+        ]
+    }
+
+    // MARK: - Test Configuration
+
+    /** Set up for each test by loading text files and instantiating the service. */
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+        instantiatePersonalityInsights()
+        loadMobyDickIntro()
+        loadKennedySpeech()
+    }
+
+    /** Instantiate Personality Insights. */
+    func instantiatePersonalityInsights() {
+        let username = Credentials.PersonalityInsightsV3Username
+        let password = Credentials.PersonalityInsightsV3Password
+        personalityInsightsV3 = PersonalityInsights(username: username, password: password)
+    }
+    
+    /** Load "MobyDickIntro.txt". */
+    func loadMobyDickIntro() {
+        let bundle = Bundle(for: type(of: self))
+        guard let file = bundle.path(forResource: "MobyDickIntro", ofType: "txt") else {
+            XCTFail("Unable to locate MobyDickIntro.txt file.")
+            return
+        }
+
+        mobyDickIntro = try? String(contentsOfFile: file)
+        guard mobyDickIntro != nil else {
+            XCTFail("Unable to read MobyDickIntro.txt file.")
+            return
+        }
+    }
+
+    /** Load "KennedySpeech.txt." */
+    func loadKennedySpeech() {
+        let bundle = Bundle(for: type(of: self))
+        guard let file = bundle.path(forResource: "KennedySpeech", ofType: "txt") else {
+            XCTFail("Unable to locate KennedySpeech.txt file.")
+            return
+        }
+
+        kennedySpeech = try? String(contentsOfFile: file)
+        guard kennedySpeech != nil else {
+            XCTFail("Unable to read KennedySpeech.txt file.")
+            return
+        }
+    }
+
+    /** Fail false negatives. */
+    func failWithError(error: Error) {
+        XCTFail("Positive test failed with error: \(error)")
+    }
+
+    /** Fail false positives. */
+    func failWithResult<T>(result: T) {
+        XCTFail("Negative test returned a result.")
+    }
+
+    /** Wait for expectations. */
+    func waitForExpectations() {
+        waitForExpectations(timeout: timeout) { error in
+            XCTAssertNil(error, "Timeout")
+        }
+    }
+    
+    // MARK: - Positive Tests
+    
+    /** Analyze the text of Kennedy's speech. */
+    func testProfile() {
+        let description = "Analyze the text of Kennedy's speech."
+        let expectation = self.expectation(description: description)
+        
+        personalityInsightsV3.getProfile(version: version,
+                                         fromText: kennedySpeech,
+                                         failure: failWithError) { profile in
+                                            for preference in profile.personality {
+                                                XCTAssertEqual("Openness", preference.name)
+                                                break
+                                             }
+                                            expectation.fulfill()
+        }
```

Comment:
Stylistically, I think there are ways that this code could be a bit more readable. When there are a lot of parameters, I tend to put each on a new-line and use an opening bracket on its own line. Here's an example:

```swift
personalityInsights.getProfile(
    version: version,
    fromText: kennedySpeech,
    failure: failWithError)
{
    profile in
    ...
}
```

It's still awkward, but it avoids all of that dead white-space from indentation.